lib/spack/spack/cmd/clean.py

```diff
@@ -97,14 +98,15 @@ def clean(parser, args):
 
     if args.python_cache:
         tty.msg('Removing python cache files')
-        for root, dirs, files in os.walk(spack_root):
-            for f in files:
-                if f.endswith('.pyc') or f.endswith('.pyo'):
-                    fname = os.path.join(root, f)
-                    tty.debug('Removing {0}'.format(fname))
-                    os.remove(fname)
-            for d in dirs:
-                if d == '__pycache__':
-                    dname = os.path.join(root, d)
-                    tty.debug('Removing {0}'.format(dname))
-                    shutil.rmtree(dname)
+        for folder in [lib_path, var_path]:
```

Comment:
I actually like `directory` better than `folder` if you're going to change the name. The docs for `os.walk` call it `top` if you want to be inline with that.